README
What is MythBox?
================
MythBox is a MythTV frontend for XBMC. 

Features:
  - Watch recordings with commercial skipping
  - Watch Live TV
  - Create and update recording schedules
  - View upcoming recordings 
  - Enhanced program guide
  - TV show and movie metadata (fanart, episodes, etc) downloaded from tvdb, tvrage, tmdb, imdb, and google image search
  - View tuner and job queue info
  - Compatible with MythTV 0.21, 0.22, and 0.23
  - Add your own public Twitter feeds
  - Move comm flagging jobs to the front of the queue

Requirements
================
  
  All Platforms
    - A working MythTV backend
    - MythTV recordings directory(s) and/or storage groups accessible via the local 
      filesystem (the machine running XBMC) pre-mounted using nfs, samba, afp, etc.   

  Linux
    - XBMC Camelot 9.11 running a recent 32-bit or 64-bit distro (Ubuntu, Debian, Fedora, Gentoo, Arch, OpenSuse, etc) 
    - ffmpeg installed as /usr/bin/ffmpeg (Ubuntu/Debian users: sudo apt-get install ffmpeg)

  Mac OSX
    - XBMC Camelot 9.11 running on Mac OSX 10.5 (Leopard) - Intel only 

  Windows
    - XBMC Camelot 9.11 running on Windows XP, Vista, or Windows 7 (32-bit or 64-bit)
  
  MythBox does *not* work on the XBox.
  
  
Installation
================
  
  1. Extract mythbox.tar.gz to the following directory:
     Linux   : ~/.xbmc/scripts 
     Mac     : ~/Library/Application Support/XBMC/scripts
     Windows : C:\Documents and Settings\[user]\Application Data\XBMC\scripts
     Vista   : C:\Users\[user]\AppData\Roaming\XBMC\scripts
           
  2. Launch XBMC
  
  3. From the main XBMC screen, select Scripts
  
  4. From the Scripts screen, select MythBox 
  
  5. You'll be sent straight to the Settings screen (with an error message). 
     Fill in all the settings and click on the 'Test Settings' button until 
     you get a 'Settings OK' message.
  
  6. Exit the Settings screen & enjoy!


Keybindings/Remote Control
============================================================================
Context         Keyboard       MCE Remote     Action
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Global          \              N/A            Toggle fullscreen/windowed mode  

Navigation      Esc            -              Go back to the previous screen
                Backspace      Back/Previous  Go back to the previous screen 
                M              Win Button     Go back to the main menu

Playing Video   X              Stop           Stop video
                Space          Pause          Pause video
                PrtScrn        Record         Take screenshot
                I              Info           Show program info                               
                O              Guide          Show audio/video/fps stats
                Z              OK             Toggle zoom modes
                Tab            N/A            Toggle fullscreen/embedded player mode

Home Window    Context Menu    TV Guide Btn   Activate popup menu to delete
Coverflow                                     or re-record the selected
                                              recording

Recording      PgUp           Channel Up      Show previous recording
Details Window PgDown         Channel Dn      Show next recording

TV Guide       PgUp           Channel Up      Scroll TV guide up one page
               PgDown         Channel Dn      Scroll TV guide down one page
               R              FastRewind      Scroll TV guide left one page
               F              FastForward     Scroll TV guide right one page                                                    
               ?              End             Scroll TV guide forward 1 day
               ?              Begin           Scroll TV guide back 1 day  
----------------------------------------------------------------------------


TODO
=============================
- Confluence skin (any skinners out there?)

History
==========================
MythBox was forked from xbmc-mythtv (http://xbmc-mythtv.sourceforge.net) 
and heavily modified to bring MythTV protocol support up to date, 
add new features, and make MythTV a pleasure to use in XBMC (obviously, not there yet).
Many thanks to all the developers whose time and effort went into the original 
xbmc-mythtv project!


Contact Info
==========================
Email      : analogue@yahoo.com
Bugs       : Create a defect @ http://mythbox.googlecode.com
Suggestions: Create an enhancement request @ http://mythbox.googlecode.com
Discuss    : http://xbmc.org/forum/showthread.php?t=43115


Acknowledgements
==========================
Thanks to Team XBMC for an awesome mediacenter!
Thanks to the MythTV devs!
Thanks to the folks at www.iconshock.com for the TV icon.


CHANGELOG
xx/xx/2010 - MythBox 1.0.0
=============================
  New
    - Issue 125 - Added Frech translations (thanks ddekani)
   
  Fixed
    - 
    

10/11/2010 - MythBox RC2
=============================
  New  
    - Added Sort feature to the Upcoming Recordings screen
    - Issue 93 - Added splash screen
    - FFmpeg output is cached for framerate extraction used by comm flagging
    - Added 'Enable Aggressive Caching' to Settings->MythTV to pre-cache framerate and commercial break info
    - Added Season and Episode to the Recording Details screen when available
    - Added TVRage as a provider of TV show metadata (used for season & episode)
    - Commercial breaks are only skipped once during playback (restarting playback in 'Play' only mode is no longer necessary if the comm break info is incorrect)
    - Issue 97 - Myth TV hostname and port are now queried directly from the database. Changed to read-only on the Settings->MythTV screen
    - Issue 106 - More responsive fanart lookup based on the current list selection in the Watch TV screen
    - Issue 109 - Enabled navigation from the MythBox Settings screen to the XBMC Settings screen 
    - Issue 114 - Added slave backend hostname to the comm flag status on the Home screen
    - Issue 118 - MythTV 0.23-1 compatibility (Protocol 57)
    - Issue 121 - German translation updates (thanks to linuxluemmel)
     
  Fixed
    - Issue 96 - Translations updated (ready for non-English contributions)
    - Issue 98 - IP addresses are used throughout the codebase. No more host not found errors.
    - Issue 107 - Database connections are now closed when idle and reconnect on demand. 
    - Issue 108 - Focus changes to Play+Skip button in Recording Details screen after user has already switched focus to another button
    - Issue 117 - Selecting the currently playing program in the TV Guide launched the Create Schedule window instead of starting Live TV 
    - Last focused recording was not restored correctly after a Refresh in the Recordings screen
 
 
06/08/2010 - MythBox RC1
=============================
 - new     : MythTV 0.23 support added - Protocol version 56 
 - new     : PageUp/PageDn (Channel Up/Channel Down on the remote) in the Recording Details screen now navigates
             to the next or previous recording
 - new     : Automatic forwarding through a commercial break when you land in the middle of the commercial break
 - new     : Added selected position and total number of items to the Recordings, Schedules, and Upcoming Recordings screens
 - new     : Selecting a recording in the Upcoming Recordings screen launches the Recording Schedule editor
 - new     : UI responsiveness tweaked to run well on an Acer Aspire Revo (1.6Ghz Atom + NVidia ION GPU) 
 - new     : TV Guide screen is back 
             PgUp/PgDown keys now scroll up/down an entire page
             Clicking on a currently playing program starts Live TV
 - new     : Program cells in the TV guide have an HD overlay indicator when appropriate
 - new     : Added keybindings and remote control buttons section to the README
 - new     : Non-idle tuners bubble to the top of the tuners list when there are more than two tuners
 - updated : Higher resolution thumbnails in Recording Details screen
 - updated : Replaced autoexpire flag with original air date on the Recording Details screen
 - updated : Faster fanart lookup on all screens + wait indicators
 - updated : Library updates: tvdb_api 1.5, IMDbPy 4.5.1  
 - updated : After deleting a recording from the Recording Details window the Recordings window would take quite a 
             while to update for a large number of recordings
 - fixed   : Requesting a recording's thumbnail from a slave backend didn't work              
 - fixed   : UI tweaks when the active skin is confluence (radio button color, active selection in coverflow, etc)
 - fixed   : Unnecessary refresh of the program listings on the Watch TV screen after stopping live tv.
 - fixed   : XBMC crashes occasionally when exiting from MythBox
 - fixed   : Clicking on Edit Schedule in the Recording Details screen when the schedule no longer exists how shows an error message
 - fixed   : Attempting to watch live tv on tuner which is already recording on a slave backend no longer fails silently
 
01/17/2010 - MythBox SVN 1797
=============================
 - new     : Watch Recordings screen rewritten in WindowXML
 - wip     : TV Guide screen temporarily disabled -- work in progress
 - new     : Home screen coverflow now supports a context menu from which you can delete or re-rerecord the recording
 - new     : Twitter feed of mythbox news events added to the bottom of the home screen
 - new     : Public twitter feeds can be added to the news feed from Settings -> Advanced -> Twitter
 - updated : Codebase purged of all legacy windowing/skinning code (non-WindowXML)
 - fixed   : File cache thread safety issues fixed
 - new     : All screens with listboxes now restore the last selected listitem
 - fixed   : Unnecessary induced reject on all but the first myth connection negotiation   
 - updated : The 'x' in the mythbox logo fixed to match the xbmc logo (tnx Jezz_X)
 - fixed   : Home screen coverflow images scaled instead of stretched
 - fixed   : Delete recording thumbnail from cache when recording deleted or marked for re-record
 - updated : Added Q15 to the FAQ - Connect to MySQL failed: 1156 (08S01): Got packets out of order,10000)
 - updated : IMDbPy updated to download higher resolution box covers (4.5)
 - new     : Tuners table on home screen now shows the time of the next scheduled recording if the tuner is currently idle
 - fixed   : Cover flow on Home screen would not get updated when a recording was deleted from the Recording Details screen
 - fixed   : Issue 70 - recordings not visible in Recording Schedules screen
 - fixed   : Issue 79 - extra set of quotes needed in call to os.system(...) on Windows (tschutte)

resources/src/msg.py
#
#  MythBox for XBMC - http://mythbox.googlecode.com
#  Copyright (C) 2009 analogue@yahoo.com
# 
#  This program is free software; you can redistribute it and/or
#  modify it under the terms of the GNU General Public License
#  as published by the Free Software Foundation; either version 2
#  of the License, or (at your option) any later version.
#
#  This program is distributed in the hope that it will be useful,
#  but WITHOUT ANY WARRANTY; without even the implied warranty of
#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#  GNU General Public License for more details.
#
#  You should have received a copy of the GNU General Public License
#  along with this program; if not, write to the Free Software
#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
#

#
# Localized string constants see strings.xml
#

class M(object):
    SETTINGS_SAVED = 25
    INFO = 26
    ERROR = 27
    CONFIRM = 28
    LOSE_CHANGES = 29
    INVALID_INT = 30
    CONNECT_MYTHTV_FAILED = 31
    CONNECT_MYSQL_FAILED = 32
    SETTINGS_OK = 33
    SETTING_DATED = 36

#  <string id="0">Myth TV</string>
#  <string id="1">Watch TV</string>
#  <string id="2">Watch Recordings</string>
#  <string id="3">TV Guide</string>
#  <string id="4">Whats Next?</string>
#  <string id="5">Status</string>
#  <string id="6">Settings</string>
#  <string id="7">Primary Host IP/Name</string>
#  <string id="8">Port</string>
#  <string id="9">Status Port</string>
#  <string id="10">Protocol Version</string>
#  <string id="11">Database Name</string>
#  <string id="12">User</string>
#  <string id="13">Password</string>
#  <string id="14">Recorded Show Prefix</string>
#  <string id="15">Live TV Prefix</string>
#  <string id="16">Local Copy Path</string>
#  <string id="17">Save</string>
#  <string id="18">Myth TV - Settings</string>
#  <string id="19">Myth TV - MySQL Settings</string>
#  <string id="20">Myth TV - Path Settings</string>
#  <string id="21">Myth TV</string>
#  <string id="22">MySQL</string>
#  <string id="23">Paths</string>
#  <string id="24">Test Settings</string>
#  <string id="25">Settings saved successfully.</string>
#  <string id="26">Information</string>
#  <string id="27">Error</string>
#  <string id="30">Invalid integer value specified.</string>
#  <string id="31">Failed to connect to primary Myth TV backend.</string>
#  <string id="32">Failed to connect to MySQL database.</string>
#  <string id="33">Settings tested successfully.</string>
#  <string id="34">Missing '%s' tag from settings.</string>
#  <string id="35">Settings file not found.</string>
#  <string id="37">MySQL Host IP/Name</string>
#  <string id="38">View By: Date Asc</string>
#  <string id="39">View By: Title Asc</string>
#  <string id="40">Title</string>
#  <string id="41">Channel</string>
#  <string id="42">Orig Air</string>
#  <string id="43">Air Date</string>
#  <string id="44">Description</string>
#  <string id="45">Unknown</string>
#  <string id="46">Not Available</string>
#  <string id="47">Recorded Show Details</string>
#  <string id="48">Play</string>
#  <string id="49">Save</string>
#  <string id="50">Expire</string>
#  <string id="51">Copy</string>
#  <string id="52">Delete</string>
#  <string id="53">Refresh</string>
#  <string id="54">Unable to find show '%s' on '%s'.</string>
#  <string id="55">Min Ring Buffer</string>
#  <string id="56">Category:</string>
#  <string id="57">Autoexpire:</string>
#  <string id="58">Yes</string>
#  <string id="59">No</string>
#  <string id="60">File Size:</string>
#  <string id="61">KB</string>
#  <string id="62">MB</string>
#  <string id="63">GB</string>
#  <string id="64">bytes</string>
#  <string id="65">Delete the show permanently?</string>
#  <string id="66">File copy complete.</string>
#  <string id="67">File copy failed.</string>
#  <string id="68">Copy the show to the local hard drive?</string>
#  <string id="69">View By: Date Desc</string>
#  <string id="70">View By: Title Desc</string>
#  <string id="71">Encoder Status</string>
#  <string id="72">Schedule</string>
#  <string id="73">Machine Information</string>
#  <string id="74">Enter the FQDN or IP address of your primary Myth TV server.</string>
#  <string id="75">Unable to parse status information returned by server.</string>
#  <string id="76">Space Free</string>
#  <string id="77">Total Space</string>
#  <string id="78">Space Used</string>
#  <string id="79">1 Minute</string>
#  <string id="80">5 Minutes</string>
#  <string id="81">15 Minutes</string>
#  <string id="82">Disk Space</string>
#  <string id="83">Load Average</string>
#  <string id="84">The next %d shows that are scheduled for recording.</string>
#  <string id="85">Unable to retrieve screenshot from backend.</string>
#  <string id="86">Your settings are out of date or missing. Please review your settings.</string>
#  <string id="87">Notes:</string>
#  <string id="88">This showing was recorded but was deleted before recording was completed.</string>
#  <string id="89">This showing was recorded but was stopped before recording was completed.</string>
#  <string id="90">This showing was recorded.</string>
#  <string id="91">This showing is being recorded.</string>
#  <string id="92">This showing will be recorded.</string>
#  <string id="93">The status of this showing is unknown.</string>
#  <string id="94">This showing was manually set to not record.</string>
#  <string id="95">This episode was previously recorded according to the duplicate policy chosen for this title.</string>
#  <string id="96">This episode was previously recorded and is still available in the list of recordings.</string>
#  <string id="97">This episode will be recorded at an earlier time instead.</string>
#  <string id="98">Too many recordings of this program have already been recorded.</string>
#  <string id="99">This was scheduled to be recorded but was manually cancelled.</string>
#  <string id="100">Another program with a higher recording priority will be recorded.</string>
#  <string id="101">This episode will be recorded at a later time instead.</string>
#  <string id="102">This showing is a repeat and will not be recorded.</string>
#  <string id="103">This is covered by another scheduled recording for the same program.</string>
#  <string id="104">There wasn't enough disk space available to record this program.</string>
#  <string id="105">The tuner card was already being used when this program was scheduled to be recorded.</string>
#  <string id="106">Do you want to automatically skip commercials?</string>
#  <string id="107">TV Guide</string>
#  <string id="108">No Data</string>
#  <string id="109">Protocol version error: %s</string>
#  <string id="110">Season Passes</string>
#  <string id="111">Once</string>
#  <string id="112">Daily</string>
#  <string id="113">Channel</string>
#  <string id="114">Always</string>
#  <string id="115">Weekly</string>
#  <string id="116">Find One</string>
#  <string id="117">Override</string>
#  <string id="118">Don't Record</string>
#  <string id="119">Find Daily</string>
#  <string id="120">Find Weekly</string>
#  <string id="121">Recording Schedule Details - "%s"</string>
#  <string id="122">Recording schedule no longer exists.</string>
#  <string id="123">Type</string>
#  <string id="124">Profile</string>
#  <string id="125">Group</string>
#  <string id="126">Priority</string>
#  <string id="127">Duplicates In</string>
#  <string id="128">Check Method</string>
#  <string id="129">Flag Commercials</string>
#  <string id="130">Auto Expire</string>
#  <string id="131">Record New Expire Old</string>
#  <string id="132">Number To Keep</string>
#  <string id="133">Start Offset</string>
#  <string id="134">End Offset</string>
#  <string id="135">Record only this showing</string>
#  <string id="136">Record daily at this time</string>
#  <string id="137">Record any time on channel %s</string>
#  <string id="138">Record any time on any channel</string>
#  <string id="139">Record weekly at this time</string>
#  <string id="140">Find and record one showing</string>
#  <string id="141">Record this showing (Override)</string>
#  <string id="142">Don't record this showing (Override)</string>
#  <string id="143">Record once each day</string>
#  <string id="144">Record once each week</string>
#  <string id="145">None</string>
#  <string id="146">Subtitle</string>
#  <string id="147">Description</string>
#  <string id="148">Subtitle and Description</string>
#  <string id="149">Current recordings</string>
#  <string id="150">Previous recordings</string>
#  <string id="151">Both current and previous</string>
#  <string id="152">Only new episodes</string>
#  <string id="153">All recordings</string>
#  <string id="154">Value must be &gt;= to %d</string>
#  <string id="155">Value must be &lt;= to %d</string>
#  <string id="156">Select a value</string>
#  <string id="157">Delete the schedule permanently?</string>
#  <string id="158">Schedule changes saved successfully.</string>
#  <string id="159">Rec Group</string>
#  <string id="160">Show Title</string>
#  <string id="161">Episode</string>
#  <string id="162">LiveTV Tuning Wait</string>
#  <string id="163">Start Page ID</string>
#  <string id="164">DVDPlayer for LiveTV</string>
#  <string id="165">DVDPlayer for Recorded</string>
#  <string id="166">Record LiveTV</string>
#  <string id="167">Record LiveTV Length</string>
#  <string id="168">Re-Record</string>
#  <string id="169">Play+Skip</string>
